### PR TITLE
Add linear elasticity example

### PR DESCRIPTION
**Pull request by larsson4**
This PR will add a linear elasticity example based on the DG linear elasticity in MFEM.

**Core functionality**
- [x] Class constructor
- [x] Class destructor
- [x] InitVariables();
- [ ] SetParameterizedProblem(problem);
- [x] AddBCFunction(dbc4, 4);
- [x] AddRHSFunction(1.0);
- [x] BuildOperators();
  - [x] InterfaceDGElasticityIntegrator
- [x] SetupBCOperators();
- [x] Assemble();
- [x] Solve();

### CODE REVIEW

File: bin/main.cpp [hunk 1]
```diff
@@ -32,6 +32,8 @@ int main(int argc, char *argv[])
       else if (mode == "build_rom")    BuildROM(MPI_COMM_WORLD);
       else if (mode == "train_rom")    TrainROM(MPI_COMM_WORLD);
       else if (mode == "single_run")   double dump = SingleRun(MPI_COMM_WORLD, output_file);
+      else if (mode == "test")    TEMPRunAndCompare();
```
Comment (dreamer2368) [inline bin/main.cpp]: I'd remove `TEMPRunAndCompare` and `OutputOperators`. They're mostly for testing/debugging purposes and don't suit the use for the main driver.

File: include/interfaceinteg.hpp [hunk 2]
```diff
@@ -11,191 +11,237 @@
...
-                                       Vector &elvect1, Vector &elvect2);
-
-   virtual void AssembleInterfaceGrad(const FiniteElement &el1,
-                                      const FiniteElement &el2,
-                                      FaceElementTransformations &Tr1,
-                                      FaceElementTransformations &Tr2,
-                                      const Vector &elfun1, const Vector &elfun2,
-                                      Array2D<DenseMatrix*> &elmats);
-};
+   class InterfaceNonlinearFormIntegrator : virtual public HyperReductionIntegrator
```
Comment (dreamer2368) [inline include/interfaceinteg.hpp:14]: Although we don't enforce any stylization yet, we'd like to keep the current indentation style. This is consistent with MFEM. Please recover the indentation throughout the code.

File: include/main_workflow.hpp [hunk 3]
```diff
@@ -24,5 +24,7 @@ void TrainROM(MPI_Comm comm);
 void AuxiliaryTrainROM(MPI_Comm comm);
 // return relative error if comparing solution.
 double SingleRun(MPI_Comm comm, const std::string output_file = "");
+void TEMPRunAndCompare();
```
Comment (dreamer2368) [inline include/main_workflow.hpp]: I'd remove `TEMPRunAndCompare` and `OutputOperators`, or at least separate them as independent executables, presumably in `sketches` directory.

File: src/interface_form.cpp [hunk 4]
```diff
@@ -10,473 +10,482 @@ using namespace std;
 namespace mfem
 {
 
-InterfaceForm::InterfaceForm(
-   Array<Mesh *> &meshes_, Array<FiniteElementSpace *> &fes_, TopologyHandler *topol_)
-   : meshes(meshes_), fes(fes_), topol_handler(topol_), numSub(meshes_.Size())
-{
-   assert(fes_.Size() == numSub);
+   InterfaceForm::InterfaceForm(
```
Comment (dreamer2368) [inline src/interface_form.cpp]: Please recover the entire indentation.

File: src/interfaceinteg.cpp [hunk 5]
```diff
@@ -12,969 +12,1279 @@ using namespace std;
...
-      for (int j = 0; j < 2; j++) assert(elmats(i, j));
-
-   bool boundary = false;
-
-   int dim, ndof1, ndof2, ndofs;
-   bool kappa_is_nonzero = (kappa != 0.);
-   double w, wq = 0.0;
-
-   dim = el1.GetDim();
+   void InterfaceNonlinearFormIntegrator::AssembleInterfaceVector(
```
Comment (dreamer2368) [inline src/interfaceinteg.cpp]: Indentation

**Review by dreamer2368 — CHANGES_REQUESTED** [review]
@larsson4 , I tried to review the changes. You've changed indentation style of a huge chunk of the existing code, making it difficult to check what has been newly added/changed in this PR.

Although we do not enforce a stylization, we've been keeping a (mostly) consistent indentation style with MFEM. Please recover all the indentations as before (as much as possible).

File: bin/main.cpp (hunk 1 above)
Comment (larsson4) [reply bin/main.cpp]: Done now! Thanks, I totally forgot about these.

**Comment by larsson4** [comment]
@dreamer2368 Indentation should be fixed now. Do you usually have a formatting file or do you keep track manually in the PR?

I tend to just use `alt+shift+f` in VSCode and that's what has given the wrong formatting

File: src/linelast_solver.cpp [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+
+   delete globalMat_mono;
+   delete globalMat;
+   delete globalMat_hypre;
+   delete mumps;
+}
+
+void LinElastSolver::SetupIC(std::function<void(const Vector &, Vector &)> F)
+{
+   init_x = new VectorFunctionCoefficient(dim, F);
```
Comment (dreamer2368) [inline src/linelast_solver.cpp:63]: Where do you delete `init_x`? I think this should be deleted in `~LinElastSolver`

**Review by dreamer2368 — APPROVED** [review]
I left a comment about `init_x` in `LinElastSolver`. I think this comment can be reflected in the next PR, so approving and merging this.

**Pull request merged**